multicluster/controllers/multicluster/label_identity_controller.go

```diff
@@ -0,0 +1,305 @@
+		WithOptions(controller.Options{
+			MaxConcurrentReconciles: common.DefaultWorkerCount,
+		}).
+		Complete(r)
+}
+
+// namespaceMapFunc enqueues Namespace events to the reconciler.
+func namespaceMapFunc(o client.Object) []reconcile.Request {
+	return []reconcile.Request{
+		{
+			NamespacedName: types.NamespacedName{
+				Name:      o.GetName(),
+				Namespace: o.GetNamespace(),
+			},
+		},
+	}
+}
+
+// onNamespaceCreateOrUpdate updates the label identities cached for all Pods in
+// the Namespace in case the Namespace's labels change
+func (r *LabelIdentityReconciler) onNamespaceCreateOrUpdate(ctx context.Context, namespace string, nsLabels map[string]string) error {
+	var pods v1.PodList
+	if err := r.Client.List(ctx, &pods, &client.ListOptions{Namespace: namespace}); err != nil {
+		klog.Errorf("Failed to list Pods in Namespace %s", namespace)
+		return err
+	}
+	for _, p := range pods.Items {
+		podKey, updatedLabel := p.Namespace+"/"+p.Name, getNormalizedLabel(nsLabels, p.Labels)
+
+		klog.V(4).InfoS("Re-processing Pod for label identity sync due to Namespace label updates", "pod", podKey)
+		if err := r.onPodCreateOrUpdate(ctx, podKey, updatedLabel, p.Spec.HostNetwork); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// onPodDelete removes the Pod and label identity mapping from the cache, and
+// updates label identity ResourceExport if necessary (the Pod deletion event
+// causes a label identity to no longer being present in the cluster).
+func (r *LabelIdentityReconciler) onPodDelete(ctx context.Context, podKey string) error {
+	// For label identity, Pod deletion has the same logic as changing a Pod to
+	// HostNetwork. So pass hostNetwork=true here.
+	_, labelToDelete := r.getLabelToUpdate(podKey, "", true)
+	if labelToDelete != "" {
+		if err := r.deleteLabelIdentityResourceExport(ctx, labelToDelete); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// onPodCreateOrUpdate updates the Pod and label identity mapping in the cache, and
+// updates label identity ResourceExport if necessary (the Pod creation/update
+// event causes a new label identity to appear in the cluster or a label identity
+// to no longer being present in the cluster).
+func (r *LabelIdentityReconciler) onPodCreateOrUpdate(ctx context.Context, podKey, normalizedLabel string, hostNetwork bool) error {
+	labelToAdd, labelToDelete := r.getLabelToUpdate(podKey, normalizedLabel, hostNetwork)
+	if labelToDelete != "" {
+		if err := r.deleteLabelIdentityResourceExport(ctx, labelToDelete); err != nil {
+			return err
+		}
+	}
+	if labelToAdd != "" {
+		if err := r.addLabelIdentityResourceExport(ctx, labelToAdd); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *LabelIdentityReconciler) getLabelToUpdate(podKey, normalizedLabel string, hostNetwork bool) (labelToAdd string, labelToDelete string) {
+	r.labelMutex.RLock()
+	defer r.labelMutex.RUnlock()
+
+	addLabel := func(podKey, normalizedLabel string) string {
```

Comment:
It will fall into the block belew, which I think is quite early?
```
if labelCacheExists && hostNetwork {
    // Non-HostNetwork Pod update to a HostNetwork Pod.
    return "", deleteLabel(podKey, oldNormalizedLabel)
}
```
What your suggest for early return?

Reply:
Code is stale